### PR TITLE
Changes for version 1.1.8 and issue 023

### PR DESCRIPTION
**Pull request by avidrucker**
- added missing syntax highlighting for label argument offsets (integer and hexadecimal)
- relaxed some rules to permit hexadecimal offsets, leading +- operators/signs, and white space padding
- wrote initial contributions doc
- added missing .orig directive (alternative version of .org)
- added syntax highlighting for hexadecimal arguments to .org and .orig
- created test file to demonstrate the full range of syntax highlighting and valid syntax in LCC Assembly: https://github.com/avidrucker/LCCextensionTesting/blob/main/happy-path.a

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Contributing to LCC-Tools for VSCode
```
Comment (lettucegoblin) [inline CONTRIBUTING]: Please change this to CONTRIBUTING.md

File: README.md [hunk 2]
```diff
@@ -31,4 +31,4 @@ The Low Cost Computer is made by Anthony J. Dos Reis for his [textbook](https://
 - **Calculate binary value to decimal & hex**
 
 # Contributing
-Feel free to contribute to this project here [on Github](https://github.com/lettucegoblin/vscode-lcc). Also feel free to fork and make your own version of this extension. Have a good day. 
+Feel free to contribute to this project here [on Github](https://github.com/lettucegoblin/vscode-lcc). Also feel free to fork and make your own version of this extension. See our [CONTRIBUTING](./CONTRIBUTING) doc for more info. Have a good day.
```
Comment (lettucegoblin) [inline README.md]: I believe this needs to be changed to CONTRIBUTING.md as well here

File: src/rules.json [hunk 3]
```diff
@@ -397,37 +397,37 @@
         },
         {
             "name": "bad operand for .word directive (integer)",
-            "pattern": "\\.word[\\t ]+(-?\\d+)\\b",
-            "validPattern": "^(-?\\d+|0x[a-fA-F0-9]+|'\\\\.'|''|'.'|[a-zA-Z_$@][a-zA-Z0-9_$@]*[ ]?[+-]?[ ]?[-?\\d+]?)$",
+            "pattern": "\\.word[\\t ]+([-+]?\\d+)\\b",
+            "validPattern": "^([-+]?\\d+|0x[a-fA-F0-9]+|'\\\\.'|''|'.'|[a-zA-Z_$@][a-zA-Z0-9_$@]*(?:[\\t ]?[+-][\\t ]?(?:\\d+))?)$",
```
Comment (lettucegoblin) [inline src/rules.json]: For whole file
[\\t ]?       -> [\\t ]*
[+-][\\t ]? -> [+-][\\t ]*

File: src/rules.json [hunk 4]
```diff
@@ -397,37 +397,37 @@
...
             "message": "The operand of '.word' must be an integer, hexadecimal (max of 4 digits), char, or label (w/ optional offset), but got {follower} (code 56 \"bad hex\")",
             "severity": "Error"
         },
         {
             "name": "bad operand for .word directive (non-number-starting catch-all)",
-            "pattern": "\\.word[\\t ]+(?!-?\\d)(?![^'\";\\t\\n\\r ]+['\"])([^';,\"\\t\\n\\r ]+)",
-            "validPattern": "^(-?\\d+|0x[a-fA-F0-9]+|'\\\\.'|''|'.'|[a-zA-Z_$@][a-zA-Z0-9_$@]*[ ]?[+-]?[ ]?[-?\\d+]?)$",
+            "pattern": "\\.word[\\t ]+(?![-+]?[ ]*\\d)(?![^'\";\\t\\n\\r ]+['\"])([^';,\"\\t\\n\\r\\-+ ]+(?:[\\t ]*[-+]?[\\t ]*[^';,\"\\t\\n\\r ]+))",
+            "validPattern": "^([\\*](?:[\\t ]*[+-][\\t ]*(?:(?:0x[a-fA-F0-9]{1,4})|(?:\\d+)))?|[a-zA-Z_$@][a-zA-Z0-9_$@]*(?:[\\t ]*[+-][\\t ]*(?:(?:0x[a-fA-F0-9]{1,4})|(?:\\d+)))?)$",
             "message": "The operand of '.word' must be an integer, hexadecimal, char, or label (w/ optional offset), but got {follower} (code 57)",
```
Comment (lettucegoblin) [inline src/rules.json:419]: Add a json comment with a rough breakdown of this

Comment (lettucegoblin) [reply src/rules.json:419]:
```json
{
    "name": "bad operand for .word directive (non-number-starting catch-all)",
    "pattern": "\\.word[\\t ]+(?![-+]?[ ]*\\d)(?![^'\";\\t\\n\\r ]+['\"])([^';,\"\\t\\n\\r\\-+ ]+(?:[\\t ]*[-+]?[\\t ]*[^';,\"\\t\\n\\r ]+))",
    "_COMMENT_pattern": "Matches the string '.word' followed by one or more spaces or tabs. Negative lookahead that asserts the string does not start with an optional '-' or '+', followed by optional spaces, and a digit. Negative lookahead that asserts the string does not contain one or more characters that are not a quote, semicolon, space, tab, newline, or carriage return, followed by a quote. Matches one or more characters that are not a semicolon, quote, comma, space, tab, newline, or carriage return, followed by an optional space, an optional '-' or '+', an optional space, and one or more characters that are not a semicolon, quote, comma, space, tab, newline, or carriage return.",
    "validPattern": "^([\\*](?:[\\t ]*[+-][\\t ]*(?:(?:0x[a-fA-F0-9]{1,4})|(?:\\d+)))?|[a-zA-Z_$@][a-zA-Z0-9_$@]*(?:[\\t ]*[+-][\\t ]*(?:(?:0x[a-fA-F0-9]{1,4})|(?:\\d+)))?)$",
    "_COMMENT_validPattern": "Matches either a '*' followed by an optional space, an optional '+' or '-', optional whitespace, and an optional hexadecimal number or decimal number, or a valid label followed by an optional space, an optional '+' or '-', an optional space, and an optional hexadecimal number or decimal number.",
    "_COMMENT_example": "Example of a string that would match this pattern: .word *-0x1A3F or .word label+123",
    "message": "The operand of '.word' must be an integer, hexadecimal, char, or label (w/ optional offset), but got {follower} (code 57)",
    "severity": "Error"
},
```

**Review by lettucegoblin — CHANGES_REQUESTED** [review]

**Comment by avidrucker** [comment]
@lettucegoblin Requested changes have been applied :+1:

**Pull request merged**